packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -927,6 +928,73 @@ export class DetailsListBase extends BaseComponent<IDetailsListProps, IDetailsLi
       onScroll(e);
     }
   };
+
+  private _getGroupProps(detailsGroupProps: IDetailsGroupRenderProps): IGroupRenderProps {
+    const {
+      onRenderFooter: onRenderDetailsGroupFooter,
+      onRenderHeader: onRenderDetailsGroupHeader
+    } = detailsGroupProps;
+    const { adjustedColumns: columns } = this.state;
+    const selection = new Selection();
+    const groupNestingDepth = this._getGroupNestingDepth();
+    const {
+      compact,
+      selectionMode,
+      viewport,
+      checkboxVisibility,
+      getRowAriaLabel,
+      getRowAriaDescribedBy,
+      checkButtonAriaLabel,
+      checkboxCellClassName
+    } = this.props;
+
+    const detailsRowProps: IDetailsRowProps | undefined = {
```

Comment:
It's not `DetailsList`'s job to compute a full `detailsRowProps`. `DetailsList` should supply minimal data to the footer, just `columns` and maybe the original `detailsListProps` or `groupProps`. Less transformation means lower render cost and smaller code size.